src/Repository/CountRepositoryInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Equip\Data\Repository;
+
+interface CountRepositoryInterface
+{
+    /**
+     * Find record count by variable criteria
+     *
+     * @param array $criteria
+     *
+     * @return int
+     */
+    public function findCountBy(array $criteria);
```

Comment:
I think using just `countBy` would be better. Semantically, we don't typically say "let me find a count" we just say "let me count it".